.github/scripts/update_docker_versions.py

```diff
@@ -81,8 +85,12 @@ def annotate(
                     image, v_current, available = get_quayio_versions(image_details)
                 else:
                     image, v_current, available = get_dockerhub_versions(image_details)
-                stable_versions = [v for v in annotate(available, stable_only=True)]
-                v_latest = sorted(stable_versions, key=lambda v: v[1], reverse=True)[0][0]
+                # Consider only stable versions unless there are none available
+                if not (candidate_versions := [v for v in annotate(available, stable_only=True)]):
+                    print(f"No stable releases identified for {image}!")
+                    v_latest = v_current
+                else:
+                    v_latest = sorted(candidate_versions, key=lambda v: v[1], reverse=True)[0][0]
```

Comment:
I think it would be clearer to put the logic for finding the latest release into a function, or do the sort in `annotate`.

I'm guessing from this that `version.Version`s are sortable?